modules/@apostrophecms/login/ui/apos/components/TheAposLogin.vue

```diff
@@ -89,15 +97,53 @@ export default {
           required: true
         }
       ],
-      context: {}
+      requirements: getRequirements(),
+      context: {},
+      requirementProps: {},
+      fetchingRequirementProps: false
     };
   },
   computed: {
-    disabled: function () {
-      return this.doc.hasErrors;
+    loaded() {
+      return this.mounted && this.beforeCreateFinished;
+    },
+    disabled() {
+      return this.doc.hasErrors || !!this.beforeSubmitRequirements.find(requirement => !requirement.done);
+    },
+    beforeSubmitRequirements() {
+      return this.requirements.filter(requirement => requirement.phase === 'beforeSubmit');
+    },
+    activeRequirement() {
+      return (this.phase !== 'beforeSubmit') && this.requirements.find(requirement => (requirement.phase !== 'beforeSubmit') && !requirement.done);
     }
   },
-  async beforeCreate () {
+  watch: {
+    async activeRequirement() {
+      if ((this.phase === 'afterPasswordVerified') && (this?.activeRequirement?.phase === 'afterPasswordVerified') && this.activeRequirement.propsRequired) {
```

Comment:
I find this confusing myself, but I think this is checking if `this` exists, which definitely wouldn't be necessary:
`this?.activeRequirement?.phase`

Reply:
still better to use the watcher's parameter instead.

Reply:
I added one too many `?.`s.